### ISSUE TITLE
can't not derive the type of path parameter, even if I manually annotated this parameter.

### ISSUE THREAD

**Issue by Core00077** [tfranzel/drf-spectacular#270]
**Describe the bug**
I've got a path parameter in my method, and my model did not contain such field. So `drf-spectacular` can't not derive the type of this path parameter. I did annotated parameter with @extend_schema. But warning INFO still pops up. 

**To Reproduce**
My@extend_schema looks like this:
![image](https://user-images.githubusercontent.com/28624864/105811408-91046300-5fe7-11eb-8fa8-b00bdcde73cf.png)

With the output below:
```
Warning #0: could not derive type of path parameter "table_name" because model "<class 'import_data.models.DataSource'>" did contain no such field. consider annotating parameter with @extend_schema. defaulting to "string".
```

**Expected behavior**
No warning info pops up.

**Comment by Core00077** [comment]
By the way, I really want to know how to bypass some warning/error info when I have a viewsets that's not based on any serializer and model. Here's a scenario:
> I've got a Model called `DataSource`, which stores config of databases. Now I need to using such configs to get info like table_names of certain database, a preview of a certain table. I developed some APIs using a new viewset, and it worked just fine, but since I didn't have any model or serializer in this viewset, drf-spectacular suggests me to use none for serializer and queryset. So I set:
> ```
>    queryset = Model.objects.none()
>    serializer_class = None
> ```
>  and it says `AttributeError: type object 'Model' has no attribute 'objects'`
>  So now I set `queryset=DataSource.objects.none()` to avoid the error, But I'm pretty sure this is not appropriate. 

As I learn from the code, drf-spectacular mainly use queryset and serializer_class to determine the types of data. But I'm not sure how to configure types to avoid auto_schema.

**Comment by tfranzel** [comment]
hi @Core00077, sry but i cannot reproduce your issue. there must be something different here, as this is a heavily tested piece of code. explicit parameter override as you did there, will remove the warning.

there is my test: 

```python
def test_foo(no_warnings):
    class XViewset(viewsets.ViewSet):
        # queryset = SimpleModel.objects.all()
        # serializer_class = SimpleSerializer

        @extend_schema(
            parameters=[
                OpenApiParameter('model_name', str, 'path'),
                OpenApiParameter('id', str, 'path')
            ],
            responses={'200': OpenApiTypes.OBJECT}
        )
        def retrieve(self, request, model_name, pk):
            pass  # pragma: no cover

    router = routers.SimpleRouter()
    router.register('/<model_name>/x', XViewset, basename='x')
    generator = SchemaGenerator(patterns=router.urls)
    schema = generator.get_schema(request=None, public=True)
    validate_schema(schema)
```

which worked fine. looks exactly like your examples. works for both `viewset` and `genericviewset`. please provide a reproduction as i cannot see the problem like this.

> AttributeError: type object 'Model' has no attribute 'objects'

that is really strange as models usually always have a manger.

> So now I set queryset=DataSource.objects.none() to avoid the error, But I'm pretty sure this is not appropriate.

this is just a convenience thing. you don't need to use it. with `@extend_schema` and `request`, `responses`, and `parameters` you can make all those errors/warnings go away usually. i have no idea why this would not work for you.

**Comment by Core00077** [comment]
Hi! @tfranzel. Appreciated your kindly reply. I found the problem. I didn't define `table_name` in url router, but set it at lookup_field of my viewsets, since I consider it as the `id` of this viewset. Here's a reproduction of your code with a little modified.
```python
import os
import django
from drf_spectacular.generators import SchemaGenerator
from drf_spectacular.types import OpenApiTypes
from drf_spectacular.utils import extend_schema, OpenApiParameter
from drf_spectacular.validation import validate_schema
from rest_framework import viewsets, routers

os.environ.setdefault("DJANGO_SETTINGS_MODULE", "cloud.settings.dev")
django.setup()


class XViewset(viewsets.ViewSet):
    # queryset = SimpleModel.objects.all()
    # serializer_class = SimpleSerializer
    lookup_field = 'table_name'
    lookup_value_regex = '[^/]+'

    @extend_schema(
        parameters=[
            OpenApiParameter('model_name', str, 'path'),
            OpenApiParameter('id', str, 'path')
        ],
        responses={'200': OpenApiTypes.OBJECT}
    )
    def retrieve(self, request, model_name, pk):
        pass  # pragma: no cover


router = routers.SimpleRouter()
router.register(r'/x/(?P<pk>\d+)', XViewset, basename='x')
generator = SchemaGenerator(patterns=router.urls)
schema = generator.get_schema(request=None, public=True)
validate_schema(schema)

```

It prints out this warnings:
```
Warning #0: failed to obtain model through view's queryset due to raised exception. prevent this either by setting "queryset = Model.objects.none()" on the view, having an empty fallback in get_queryset() or by using @extend_schema. (Exception: 'XViewset' object has no attribute 'get_queryset')
Warning #1: could not derive type of path parameter "table_name" because because it is untyped and obtaining queryset from <class '__main__.XViewset'> failed. consider adding a type to the path (e.g. <int:table_name>) or annotating the parameter type with @extend_schema. defaulting to "string".
```

Seems to me that drf-spectacular didn't expect `lookup_field` in viewsets. 

In addtion, when setting 
>    ``` 
>  # queryset = SimpleModel.objects.all()
>  # serializer_class = SimpleSerializer 
>    ```

to 
>    ``` 
>    queryset = Model.objects.all()
>    serializer_class = None
>    ```

it raises `AttributeError: type object 'Model' has no attribute 'objects'` as I mentioned. Using a random model at `queryset` would avoid `Warning #0`. And that is where I am right now.

**Comment by tfranzel** [comment]
i had time to revisit this. i must say the example feels weird but i'll take at at face value. 

you changed `lookup_field = 'table_name'` (default: `pk`), while you route `pk` separately in the router. 

1. Because `SCHEMA_COERCE_PATH_PK` is on by default, `OpenApiParameter('id', str, 'path')` will capture the `pk`
2. you wrote `model_name` but that is wrong. the parameter must match the `lookup_field` with `OpenApiParameter('table_name', str, 'path'),`. if those 2 names match there is no problem anymore.

the url has 2 parameters and both get resolved like this, which means there is no access to queryset anymore, and no exception will be thrown.

so this works for me 
```python
    class XViewset(viewsets.ViewSet):
        # queryset = SimpleModel.objects.all()
        # serializer_class = SimpleSerializer
        lookup_field = 'table_name'
        lookup_value_regex = '[^/]+'

        @extend_schema(
            parameters=[
                OpenApiParameter('table_name', str, 'path'),
                OpenApiParameter('id', str, 'path')
            ],
            responses={'200': OpenApiTypes.OBJECT}
        )
        def retrieve(self, request, model_name, pk):
            pass  # pragma: no cover

    router = routers.SimpleRouter()
    router.register(r'/x/(?P<pk>\d+)', XViewset, basename='x')
```

**Comment by Core00077** [comment]
Holy shot, this is a mistake. I didn't make my problem reproduced right. When I corrected this demo, the warnings are gone. However, my problem still exists.

After some more reviews, I got it. The actual problem is this: I have two methods that uses "table_name", but I only configured one of them with parameters. I've been thinking that the warning refers to the first method, but it actually refers to the other one. GOSH!!

Sorry for bothering you, and thank you for your time. It would be much nicer if warnings can give some hints that shows where they came from.
